tests/suites/hup/tests/storagemigration.js

```diff
@@ -0,0 +1,458 @@
+						this,
+						test,
+						'image',
+						this.context.get().hup.payload,
+						this.context.get().link,
+					);
+
+				// TODO: this copies a large chunk of the rollback-health/openvpn
+				// test case. how useful is this? we just want to check that
+
+				// break openvpn
+				test.comment(`Breaking openvpn to trigger rollback-health...`);
+				await this.context
+					.get()
+					.worker.executeCommandInHostOS(
+						`ln -sf /dev/null $(find /mnt/sysroot/inactive/ | grep "bin/openvpn$")`,
+						this.context.get().link,
+					);
+
+				test.comment(
+					`Pretend VPN was previously active for unmanaged OS suite...`,
+				);
+				await this.context
+					.get()
+					.worker.executeCommandInHostOS(
+						`sed 's/BALENAOS_ROLLBACK_VPNONLINE=0/BALENAOS_ROLLBACK_VPNONLINE=1/' -i /mnt/state/rollback-health-variables && sync -f /mnt/state`,
+						this.context.get().link,
+					);
+
+				await rebootAndWaitForRollbackHealth(this, test);
+
+				test.is(
+					await this.context.get().worker.getOSVersion(this.context.get().link),
+					versionBeforeHup,
+					`The OS version should have reverted to ${versionBeforeHup}`,
+				);
+
+				// 0 means file exists, 1 means file does not exist
+				test.is(
+					await this.context
+						.get()
+						.worker.executeCommandInHostOS(
+							`test -f /mnt/state/rollback-health-triggered ; echo $?`,
+							this.context.get().link,
+						),
+					'0',
+					'There should be a rollback-health-triggered file in the state partition',
+				);
+
+				// 0 means directory exists, 1 means directory does not exist
+				test.is(
+					await this.context.get().worker.executeCommandInHostOS(
+						`test -d /mnt/data/docker/overlay2; echo $?`,
+						this.context.get().link,
+					),
+					'0',
+					'There should be a /mnt/data/docker/overlay2 directory, indicating that storage-migration was run.'
+				);
+
+				// check that the old storage driver is used by balena-engine
+				test.is(
+					await this.context.get().worker.executeCommandInHostOS(
+						`balena info 2>/dev/null | awk '/Storage.Driver/{print $3}'`,
+						this.context.get().link),
+					'aufs',
+					`The engine should be configured with the aufs storage driver`
+				);
+
+				const containersAfterHup = (await this.context
+					.get()
+					.worker.executeCommandInHostOS(
+						`balena container ls --format '{{.ID}}' | xargs balena inspect --format '{{.GraphDriver.Name}}'`,
+						this.context.get().link,
+					))
+					.trim()
+					.split('\n');
+
+				// check that we're still running the old containers
+				test.is(
+					containersAfterHup.length,
+					containersBeforeHup.length,
+					'All previous containers should run',
+				)
+
+				// false means at least 1 container was not configured with overlay2
+				test.is(
+					containersAfterHup.reduce((a, v) => a && (v === 'aufs'), true),
+					true,
+					`All containers should be running on the aufs storage-driver`,
+				);
+
+				await archiveMigrationLogs(this, test);
```

Comment:
This will fail if the rollback version (pre-hup) version is too old to support the `-b all` flag with journalctl. We've seen this with raspberry-pi2 since it has no recent releases.

Reply:
https://github.com/balena-os/meta-balena/pull/2303

Reply:
@klutchell +1 copied your approach from the linked pr